storyprotocol/python-sdk
# [Feature] Add Unit and Integration Tests for StoryClient and IPAsset

## Description
This PR introduces extensive testing for the StoryClient and IPAsset components of the Python SDK. Key changes include:
1. Added unit tests for StoryClient and IPAsset.
2. Refactored story_client.py to remove unnecessary code and streamline class definition.
3. Implemented integration tests for IP asset registration functionality.
4. Ensured consistent code structure and organization for better maintainability.

## Test Plan 
1. Run unit tests for StoryClient and IPAsset:
- Use pytest to execute tests in tests/unit/ directory.
- Verify the correctness of StoryClient initialization and IPAsset registration logic.

2. Run integration tests for IP asset registration:
- Use pytest to execute tests in tests/integration/ directory.
- Validate the interaction

3. Review and confirm the proper organization and refactoring of the story_client.py file.

## Related Issue
n/a

## Notes
- Ensure the environment variables (WALLET_PRIVATE_KEY and RPC_PROVIDER_URL) are set correctly in the .env file for the tests to run successfully.
